### PR TITLE
InfluxDB.Client.AspNetCore

### PR DESCRIPTION
**Pull request by deinok**
This adds integration for DependencyInjection and HealthChecks. 
The idea behind is making easy to integrate with AspNetCore but in fact is not really tied to AspNetCore so im not sure about naming it InfluxDB.Client.AspNetCore. Also, the two integrations could be splited in two diferent packages.

PD: The namespaces of the DI related extensions are 'Microsoft.Extensions.DependencyInjection' as it is the recomended.
ref: #158

### CODE REVIEW

**Comment by bednar** [comment]
@deinok Thanks for your PR 👍 We will take a look

File: InfluxDB.Client.AspNetCore/InfluxDB.Client.AspNetCore.csproj [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
```
Comment (bednar) [inline InfluxDB.Client.AspNetCore/InfluxDB.Client.AspNetCore.csproj:5]: Please add also support for `netstandard2.1` and set `LangVersion` to `8`:

```suggestion
  <PropertyGroup>
    <TargetFrameworks>netstandard2.0;netstandard2.1</TargetFrameworks>
    <LangVersion>8</LangVersion>
  </PropertyGroup>
```

File: InfluxDB.Client.AspNetCore/InfluxDB.Client.AspNetCore.csproj [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
```
Comment (bednar) [inline InfluxDB.Client.AspNetCore/InfluxDB.Client.AspNetCore.csproj:6]: Please add a detail information about package, something like in `Client.csproj`:

```xml
<Description>...</Description>
<Authors>influxdb-client-csharp Contributors</Authors>
<AssemblyName>InfluxDB.Client.AspNetCore</AssemblyName>
<VersionPrefix>1.16.0</VersionPrefix>
<VersionSuffix>dev</VersionSuffix>

<PackageId>InfluxDB.Client.AspNetCore</PackageId>
<PackageTags>influxdata;timeseries;flux;influxdb</PackageTags>
<PackageIcon>influxdata.jpg</PackageIcon>
<PackageIconUrl>https://raw.githubusercontent.com/influxdata/influxdb-client-csharp/master/Assets/influxdata.jpg</PackageIconUrl>
<PackageProjectUrl>https://github.com/influxdata/influxdb-client-csharp/tree/master/InfluxDB.Client.AspNetCore</PackageProjectUrl>
<PackageLicenseExpression>MIT</PackageLicenseExpression>
<RepositoryUrl>https://github.com/influxdata/influxdb-client-csharp</RepositoryUrl>
<RepositoryType>git</RepositoryType>

<PackageReleaseNotes>https://github.com/influxdata/influxdb-client-csharp/blob/master/CHANGELOG.md</PackageReleaseNotes>
<PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
<RootNamespace>InfluxDB.Client.AspNetCore</RootNamespace>

<AssemblyOriginatorKeyFile>../Keys/Key.snk</AssemblyOriginatorKeyFile>
<SignAssembly>true</SignAssembly>
```

File: InfluxDB.Client.AspNetCore/InfluxDBClientHealthCheck.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+using System.Threading;
+using System.Threading.Tasks;
+using InfluxDB.Client.Api.Domain;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace InfluxDB.Client.AspNetCore
+{
+
+    public class InfluxDBClientHealthCheck : IHealthCheck
```
Comment (bednar) [inline InfluxDB.Client.AspNetCore/InfluxDBClientHealthCheck.cs:9]: Please add tests for `InfluxDBClientHealthCheck`. We want to be sure that everything works between releases.

File: InfluxDB.Client.AspNetCore/ServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+using System.ComponentModel;
+using InfluxDB.Client;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+
+    public static class ServiceCollectionExtensions
+	{
+
+		public static IServiceCollection AddInfluxDBClient(this IServiceCollection serviceCollection, string connectionString, ServiceLifetime serviceLifetime = ServiceLifetime.Scoped)
```
Comment (bednar) [inline InfluxDB.Client.AspNetCore/ServiceCollectionExtensions.cs:10]: Please add tests for `AddInfluxDBClient`. We want to be sure that everything works between releases.

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks again for your PR 👍 

There is a few requirements that should be satisfy before we accept the PR. Could you also add a short README.md that will clarify how to use `InfluxDB.Client.AspNetCore`?

**Comment by deinok** [comment]
@bednar Hey, take a quick look, but I need the connection string for CircleCI to make them pass.

**Comment by bednar** [comment]
You could extend `AbstractTest` and take InfluxDB URL by: https://github.com/influxdata/influxdb-client-csharp/blob/d1c12e19124a43b6b39514dbb41dffea0fb8e86e/Client.Core.Test/AbstractTest.cs#L60

**Comment by bednar** [comment]
@deinok, do you want to continue with this PR?

**Comment by deinok** [comment]
Project got stuck from my enterprise, for now you can close it. If they wake up again I will do it again.

**Comment by bednar** [comment]
@deinok thanks for info

**Pull request closed without merging**